### ISSUE TITLE
problem for SummaryGraph

### ISSUE THREAD

**Issue by wuzhi19931128** [NervanaSystems/distiller#436]
I have a very simple net with shortcut structure.However , there is a loop if use summarygraph。What should I do to reduce this loop?Thanks

import torch
import torch.nn as nn

class Net(nn.Module):
    def __init__(self):
        super(Net,self).__init__()
        self.conv1 = nn.Conv2d(3, 64, 7, stride=2, padding=3, groups=1)
        self.layer1_0_conv1 = nn.Conv2d(64, 64, 1, stride=1, padding=0, groups=1)
        self.top_level_op = nn.AvgPool2d(kernel_size=7)
        self.last_linear = nn.Linear(16384, 1000)

 def forward(self,x):
        x_1 = self.conv1(x)
        x_2 = self.layer1_0_conv1(x_1)
        x_4 = x_2 + x_1
        x_5 = self.top_level_op(x_4)
        x_6 = x_5.view(x_5.size(0), -1)
        x_7 = self.last_linear(x_6)
        return x_7

**Comment by levzlotnik** [comment]
Hi @wuzhi19931128 ,

Could you please add the code where you're using the `SummaryGraph`?

Thanks,
Lev

**Comment by wuzhi19931128** [comment]
Sorry,  I found the reason.  I wrote a program to general pytorch model definition from distiller summary graph. So one layer of my net is named by "top_level_op" . When Summary graph analysis 
this net , it's confused by layer and pytorch op . When I change the name , it acts normally.
